packages/kbn-config-schema/README.md

```diff
@@ -0,0 +1,461 @@
+```typescript
+const valueSchema  = schema.byteSize({ min: '3kb' });
+```
+
+__Notes:__
+* The string value for `schema.byteSize()` and its options supports the following prefixes: `b`, `kb`, `mb`, `gb` and `tb`.
+* The number value is treated as a number of bytes and hence should be a positive integer, e.g. `100` is equal to `'100b'`.
+* Currently you cannot specify zero bytes with a string format and should use number `0` instead.
+
+#### `schema.duration()`
+
+Validates input data as a proper [duration](https://momentjs.com/docs/#/durations/).
+
+__Output type:__ `moment.Duration`
+
+__Options:__
+  * `defaultValue: moment.Duration | string | number | Reference<moment.Duration | string | number> | (() => moment.Duration | string | number)` - defines a default value, see [Default values](#default-values) section for more details.
+  * `validate: (value: moment.Duration | string | number) => string | void` - defines a custom validator function, see [Custom validation](#custom-validation) section for more details.
+
+__Usage:__
+```typescript
+const valueSchema  = schema.duration({ defaultValue: '70ms' });
+```
+
+__Notes:__
+* The string value for `schema.duration()` supports the following prefixes: `ms`, `s`, `m`, `h`, `d`, `w`, `M` and `Y`.
+* The number value is treated as a number of milliseconds and hence should be a positive integer, e.g. `100` is equal to `'100ms'`.
+
+#### `schema.conditional()`
+
+Allows to specify a condition that is evaluated _at the validation time_ and results into either one or another input validation schema.
+
+The first argument is always a [reference](#references) while the second one can be:
+* another reference, in this cases both references are "dereferenced" and compared
+* schema, in this case the schema is used to validate "dereferenced" value of the first reference
+* value, in this case "dereferenced" value of the first reference is compared to that value
+
+The third argument is a schema that should be used if the result of the aforementioned comparison evaluates to `true`, otherwise `schema.conditional()` should fallback
+to the schema provided as the fourth argument.
+
+__Output type:__ `TConditional1 | TConditional2`
+
+__Options:__
+  * `defaultValue: TConditional1 | TConditional2 | Reference<TConditional1 | TConditional2> | (() => TConditional1 | TConditional2` - defines a default value, see [Default values](#default-values) section for more details.
+  * `validate: (value: TConditional1 | TConditional2) => string | void` - defines a custom validator function, see [Custom validation](#custom-validation) section for more details.
+
+__Usage:__
+```typescript
+const valueSchema  = schema.object({
+  key: schema.oneOf([schema.literal('number'), schema.literal('string')]),
+  value: schema.conditional(schema.siblingRef('key'), 'number', schema.number(), schema.string()),
+});
+```
+
+__Notes:__
+* Conditional schemas may be hard to read and understand and hence should be used only sparingly.
```

Comment:
Well, if there were no cases where we need conditional schemas we wouldn't introduce them ;) It's just that it's hard to make conditional schema useful and readable at the same time :disappointed:

Reply:
I feel the same every time I see [rxjs.iif](https://rxjs-dev.firebaseapp.com/api/index/function/iif)
Would it be useful to follow their naming convention and use `trueResult/falseResult` instead of `TConditional1/TConditional2` ?

Reply:
Yep, it's a good idea, I'll use `TTrueResult` and `TFalseResult` to be in line with other generic type names I use here.